teacher.rb

```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization:, name: 'Unknown', age: 0)
+    super(id, name, age, parent_permission: true)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Ouch! this is a close one, but since the `Teacher` class inherits from the `Person` class it's important to ensure that all the necessary parameters are included in the constructor. I noticed that the `parent_permission` parameter is missing. so kindly add the parameter at line `2` with the default value set to `true` which means you just need to pass the `parent_permission` argument without setting a default value at line `3` when calling the parent constructor.
- [ ] Since the constructor in the parent `Person` class doesn't accept an `id` as part of its argument I'd suggest removing it from the code at line `3`.